pettingzoo/butterfly/knights_archers_zombies/src/players.py

```diff
@@ -88,31 +99,42 @@ def __init__(self, agent_name):
         # self.radius = radius
         self.attacking = False  # Used to disable movement during attacking
         self.attack_phase = -5
-        self.weapon = Sword(self)
+        self.weapon = Sword(self, FPS)
         self.alive = True  # This flag is used to immediately delete the mace object when the knight dies
         self.action = -1
         self.score = 0
         self.is_archer = False
         self.is_knight = True
         self.agent_name = agent_name
 
+        self.knight_speed = KNIGHT_SPEED * 15. / FPS
+        self.angle_rate = ANGLE_RATE * 15. / FPS
+
+        if self.knight_speed % 1. != 0.:
```

Comment:
Just curious, why is it  problem if the knight speed is fractional?

Reply:
At first glance I do not think this will be an issue, but I kept the same convention to keep in line with the other butterfly envs where speeds are usually in terms of grid spacing per step. It's mostly to assert that I don't break anything else down the line.